File: 9/c++/solution.cpp
#include <stdio.h>

using namespace std;

class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0){
            return false;
        }
        if (x == 0){
            return true;
        }

        int digits[32];
        int total_digit_count = 0;

        while (x > 0){
            digits[total_digit_count] = x % 10;
            x = x /10;
            total_digit_count++;
        }

        for (int i = 0; i < total_digit_count / 2; i++){
            if (digits[i] != digits[total_digit_count - 1 - i]){
                return false;
            }
        }
        return true;
    }
};

int main(){
    int test_input = 121;
    bool wanted_result = true;

    Solution* solver = new Solution();

    if (solver->isPalindrome(test_input) == wanted_result){
        printf("Test passed\n");
    }
    else{
        printf("Test failed\n");
    }
}

File: 9/python/solution.py

class Solution:
    def isPalindrome(self, x: int) -> bool:
        num_string = str(x)
        digits = len(num_string)

        for i in range(0, digits // 2):
            if not num_string[i] == num_string[digits - 1 - i]:
                return False
        return True
    

if __name__ == "__main__":
    num = 121
    sol = Solution()
    res = sol.isPalindrome(num)

    if res:
        print("Passed")
    else:
        print("Failed")